astropy/io/fits/tests/test_connect.py

```diff
@@ -414,6 +414,64 @@ def test_mask_str_on_read(self, tmp_path):
         tab = Table.read(filename, mask_invalid=False)
         assert tab.mask is None
 
+    def test_heterogeneous_VLA_tables(self, tmp_path):
+        """
+        Check the behaviour of heterogeneous VLA object.
+        """
+        filename = tmp_path / "test_table_object.fits"
+        msg = "Column 'col1' contains unsupported object types or mixed types: "
+
+        # The column format fix the type of the arrays in the VLF object.
+        a = np.array([45, 30])
+        b = np.array([11.0, 12.0, 13])
+        var = np.array([a, b], dtype=object)
+        tab = Table({"col1": var})
+        with pytest.raises(TypeError, match=msg):
+            tab.write(filename)
+
+        # Strings in the VLF object can't be added to the table
+        a = np.array(["five", "thirty"])
+        b = np.array([11.0, 12.0, 13])
+        var = np.array([a, b], dtype=object)
+        with pytest.raises(TypeError, match=msg):
+            tab.write(filename, overwrite=True)
```

Comment:
`overwrite=True` should not be needed here since the previous `.write` raised an error.

Reply:
```suggestion
            tab.write(filename)
```